ppl/src/main/java/com/amazon/opendistroforelasticsearch/sql/ppl/utils/PPLQueryDataAnonymizer.java

```diff
@@ -0,0 +1,326 @@
+    for (Map renameMap : node.getRenameList()) {
+      renameMapBuilder.put(visitExpression(renameMap.getOrigin()),
+          ((Field) renameMap.getTarget()).getField().toString());
+    }
+    String renames =
+        renameMapBuilder.build().entrySet().stream().map(entry -> StringUtils.format("%s as %s",
+            entry.getKey(), entry.getValue())).collect(Collectors.joining(","));
+    return StringUtils.format("%s | rename %s", child, renames);
+  }
+
+  /**
+   * Build {@link LogicalAggregation}.
+   */
+  @Override
+  public String visitAggregation(Aggregation node, String context) {
+    String child = node.getChild().get(0).accept(this, context);
+    final String group = visitExpressionList(node.getGroupExprList());
+    return StringUtils.format("%s | stats %s", child,
+        String.join(" ", visitExpressionList(node.getAggExprList()), groupBy(group)).trim());
+  }
+
+  /**
+   * Build {@link LogicalRareTopN}.
+   */
+  @Override
+  public String visitRareTopN(RareTopN node, String context) {
+    final String child = node.getChild().get(0).accept(this, context);
+    List<Argument> options = node.getNoOfResults();
+    Integer noOfResults = (Integer) options.get(0).getValue().getValue();
+    String fields = visitFieldList(node.getFields());
+    String group = visitExpressionList(node.getGroupExprList());
+    return StringUtils.format("%s | %s %d %s", child,
+        node.getCommandType().name().toLowerCase(),
+        noOfResults,
+        String.join(" ", fields, groupBy(group)).trim()
+    );
+  }
+
+  /**
+   * Build {@link LogicalProject} or {@link LogicalRemove} from {@link Field}.
+   */
+  @Override
+  public String visitProject(Project node, String context) {
+    String child = node.getChild().get(0).accept(this, context);
+    String arg = "+";
+    String fields = visitExpressionList(node.getProjectList());
+
+    if (node.hasArgument()) {
+      Argument argument = node.getArgExprList().get(0);
+      Boolean exclude = (Boolean) argument.getValue().getValue();
+      if (exclude) {
+        arg = "-";
+      }
+    }
+    return StringUtils.format("%s | fields %s %s", child, arg, fields);
+  }
+
+  /**
+   * Build {@link LogicalEval}.
+   */
+  @Override
+  public String visitEval(Eval node, String context) {
+    String child = node.getChild().get(0).accept(this, context);
+    ImmutableList.Builder<Pair<String, String>> expressionsBuilder = new ImmutableList.Builder<>();
+    for (Let let : node.getExpressionList()) {
+      String expression = visitExpression(let.getExpression());
+      String target = let.getVar().getField().toString();
+      expressionsBuilder.add(ImmutablePair.of(target, expression));
+    }
+    String expressions = expressionsBuilder.build().stream().map(pair -> StringUtils.format("%s"
+        + "=%s", pair.getLeft(), pair.getRight())).collect(Collectors.joining(" "));
+    return StringUtils.format("%s | eval %s", child, expressions);
+  }
+
```

Comment:
+`visitAggregation` why not leave the names of operations and functions unmasked